.github/workflows/build-and-deploy.yml

```diff
@@ -302,30 +303,17 @@ jobs:
           path: coverage/.last_run.json
           retention-days: 5
 
-      # Workaround for https://github.com/dawidd6/action-download-artifact/issues/147
-      - name: Get Run ID
-        id: get_run_id
-        run: |
-          echo "::set-output name=run_id::$(\
-            gh run list \
-              --workflow build-and-deploy.yml \
-              --limit 100 \
-              --json conclusion,status,databaseId,headBranch \
-              --jq ". | map(select( .conclusion == \"success\" and .headBranch == \"main\")) | first(.[]).databaseId" \
-          )"
-        env:
-          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
-
       - name: Download main branch coverage artifact
         id: download-base-coverage
         uses: dawidd6/action-download-artifact@v2
         if: github.event_name == 'pull_request'
         with:
           github_token: ${{secrets.GITHUB_TOKEN}}
           workflow: build-and-deploy.yml
+          workflow_conclusion: success
+          search_artifacts: true
```

Comment:
do we need to specify the main branch again?

Reply:
@cwrw Unless it's coincidence that the later builds here passed I think not specifying makes the initial paginated search with the GitHub client code less filtered https://github.com/dawidd6/action-download-artifact/blob/master/main.js#L68 so it just gets everything and then skips unsuccessful workflow runs. When it finds a successful run it then looks for an artifact matching the specified name.
It's less performant but it presumably bypasses the filtering bug on the Github end.

Reply:
Does that mean it downloads from the `main` branch because that's the only one we upload the coverage report for?  (rather than because we explicitly pass the branch name in as a parameter).  Might be worth adding a comment if that's the case

Reply:
Yes it will only ever find an artifact with the name `base-coverage` on a successful workflow run and this will only ever be a run on the `main` branch as that's the condition for storing the coverage artifact.
This config just circumvents specifying/searching on the branch name and presumably getting an invalid run ID as a result.
Will add a comment.